components/feature/intent/src/main/java/mozilla/components/feature/intent/processing/TabIntentProcessor.kt

```diff
@@ -50,8 +50,9 @@ class TabIntentProcessor(
         return if (url.isNullOrEmpty()) {
             false
         } else {
-            val session = createSession(url, private = isPrivate, source = Source.ACTION_VIEW)
-            loadUrlUseCase(url, session, LoadUrlFlags.external())
+            sessionManager.sessions.find { it.url == url }?.also { sessionManager.select(it) }
```

Comment:
I think this is a little hard to read as a one liner. In this case it's better to break up the logic a little, something like:
```kt
val existingSession = sessionManager.sessions.find { it.url == url }
if (existingSession != null) {
  ...
} else {
  ...
}
```